### PR TITLE
MLIBZ-1673: unit test to test if the refresh token trigger enters in a infinite loop

### PR DESCRIPTION
**Pull request by heyzooi**
#### Description

Unit test to test if the refresh token trigger enters in a infinite loop

#### Changes

No changes in the code itself since the unit test proved the infinite loop is not happening

#### Tests

* Test if the refresh token trigger enters in a infinite loop

### CODE REVIEW

File: Kinvey/KinveyTests/UserTests.swift [hunk 1]
```diff
@@ -1828,5 +1828,177 @@ class UserTests: KinveyTestCase {
...
+        defer {
+            setURLProtocol(nil)
+        }
+        
+        XCTAssertNil(client.activeUser)
+        
+        weak var expectationLogin = expectation(description: "Login")
+        
+        let redirectURI = URL(string: "micAuthGrantFlow://")!
+        User.login(
```
Comment (tejasranade) [inline Kinvey/KinveyTests/UserTests.swift]: @heyzooi Shouldn't this be `User.loginWithAuthorization` as described [here](http://devcenter.kinvey.com/ios/guides/mobile-identity-connect#automated-authorization-grant)?

File: Kinvey/KinveyTests/UserTests.swift [hunk 2]
```diff
@@ -1828,5 +1828,177 @@ class UserTests: KinveyTestCase {
...
+        }
+        
+        XCTAssertNotNil(client.activeUser)
+        
+        do {
+            let store = DataStore<Person>.collection(.network)
+            
+            weak var expectationFind = expectation(description: "Find")
+            
+            store.find { persons, error in
```
Comment (tejasranade) [inline Kinvey/KinveyTests/UserTests.swift]: @heyzooi I get what's being tested here - is there a better way to test this? For example, this doesn't verify that the token refresh request got fired once and failed.

File: Kinvey/KinveyTests/UserTests.swift [hunk 3]
```diff
@@ -1828,5 +1828,180 @@ class UserTests: KinveyTestCase {
...
+                        "_acl" : [
+                            "creator" : "57c788d168d976c525ee4602"
+                        ]
+                    ] as [String : Any]
+                    let data = try! JSONSerialization.data(withJSONObject: json, options: [])
+                    client?.urlProtocol(self, didLoad: data)
+                    client?.urlProtocolDidFinishLoading(self)
+                default:
+                    type(of: self).invalidCredentialsCount += 1
+                    XCTAssertLessThanOrEqual(type(of: self).invalidCredentialsCount, 2)
```
Comment (heyzooi) [inline Kinvey/KinveyTests/UserTests.swift]: now checking if multiple request are tried after a token expires

File: Kinvey/KinveyTests/UserTests.swift (hunk 1 above)
Comment (heyzooi) [inline Kinvey/KinveyTests/UserTests.swift]: it will be changed as discussed in the documentation

File: Kinvey/KinveyTests/UserTests.swift (hunk 2 above)
Comment (heyzooi) [inline Kinvey/KinveyTests/UserTests.swift]: tested was added in the mock response... please check mock above

**Comment by heyzooi** [comment]
@tejasranade let me know if I can merge

**Pull request merged**